test/performance/odp_sched_perf.c

```diff
@@ -356,12 +372,83 @@ static int set_num_cpu(test_global_t *global)
 	return 0;
 }
 
-static int create_pool(test_global_t *global)
+static odp_pool_t create_buffer_pool(odp_pool_capability_t *pool_capa, uint32_t tot_event,
+				     uint32_t event_size)
 {
-	odp_pool_capability_t pool_capa;
 	odp_pool_param_t pool_param;
 	odp_pool_t pool;
 	uint32_t max_num, max_size;
+
+	printf("  pool type                 buffer\n");
+
+	max_num = pool_capa->buf.max_num;
+	max_size = pool_capa->buf.max_size;
+
+	if (max_num && tot_event > max_num) {
+		ODPH_ERR("Error: max events supported %u\n", max_num);
+		return ODP_POOL_INVALID;
+	}
+
+	if (max_size && event_size > max_size) {
+		ODPH_ERR("Error: max supported event size %u\n", max_size);
+		return ODP_POOL_INVALID;
+	}
+
+	odp_pool_param_init(&pool_param);
+	pool_param.type = ODP_POOL_BUFFER;
+	pool_param.buf.num = tot_event;
+	pool_param.buf.size = event_size;
+	pool_param.buf.align = 8;
+
+	pool = odp_pool_create("sched perf", &pool_param);
+
+	if (pool == ODP_POOL_INVALID)
+		ODPH_ERR("Error: buffer pool create failed\n");
+
+	return pool;
+}
+
+static odp_pool_t create_packet_pool(odp_pool_capability_t *pool_capa, uint32_t tot_event,
+				     uint32_t event_size)
```

Comment:
This is practically the same as create_buffer_pool(). Should these two be combined into one function that takes pool type as a parameter?

Reply:
Fixed in V2.